Automattic/jetpack
# CI: Hack around Github bug when skipping matrixed jobs

#### Changes proposed in this Pull Request:

Apparently Github checks the `if` before evaluating the matrix, and if the job is to be skipped it skips a single job with the `matrix` contexts unexpanded instead of skipping the expanded jobs. This makes things fail if those matrixed jobs are marked as required.
    
For now, let's hack around the problem by adding a step to the "detect changed files" job to manually apply the necessary statuses when the job is going to be skipped.

#### Jetpack product discussion


p1606835667387400-slack-CBG1CP4EN

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






* See if jobs "PHP lint (5.6)", "PHP lint (7.0)", "PHP lint (7.4)", and "PHP lint (8.0)" are running correctly and passing on this PR.
* See if "PHP lint (5.6)", "PHP lint (7.0)", "PHP lint (7.4)", and "PHP lint (8.0)" are correctly showing as passing/skipped on #17941. It's ok that they lack the "PHP Linting" prefix.

#### Proposed changelog entry for your changes:


* None needed.
